### PR TITLE
Allow formatting UnboundedQuantityValues

### PR DESCRIPTION
**Pull request by thiemowmde**
This is split from #66. I pulled all new test cases from #66, but not the implementation. Instead I tried to find the most minimal change that makes all tests succeed. Plus I added more test cases, especially for the edge case "do not add additional zeros when rounding to a specific number of decimal places". This got fixed by #69.

[Bug: T139748](https://phabricator.wikimedia.org/T139748)

### CODE REVIEW

**Comment by thiemowmde** [comment]
Rebased. Pinging @brightbyte.

File: src/ValueFormatters/QuantityFormatter.php [hunk 1]
```diff
@@ -148,6 +151,22 @@ protected function formatQuantityValue( QuantityValue $quantity ) {
...
+	 *
+	 * @return string
+	 */
+	protected function formatUnboundedQuantityValue( UnboundedQuantityValue $quantity ) {
+		$amount = $quantity->getAmount();
+		$roundingExponent = $this->options->getOption( self::OPT_APPLY_ROUNDING );
+
+		if ( !is_bool( $roundingExponent ) ) {
+			$amount = $this->decimalMath->roundToExponent( $amount, (int)$roundingExponent );
+		}
```
Comment (brightbyte) [inline src/ValueFormatters/QuantityFormatter.php]: If $roundingExponent === true, we should guess the precision, and then apply rounding.

Comment (brightbyte) [inline src/ValueFormatters/QuantityFormatter.php]: It's a bit odd that we don't apply rounding even if OPT_APPLY_ROUNDING is true...

Comment (thiemowmde) [inline src/ValueFormatters/QuantityFormatter.php]: But if all you have is the string "1.499999999999999", no bounds and an option set to `true`, how do you round?

Comment (brightbyte) [inline src/ValueFormatters/QuantityFormatter.php]: Yea, the idea is that the automatic precision guessing would guess a precision that would not change the number when rounding is applied. So it's equivalent to no rounding. Still, it feels odd.
